tf2/include/tf2/time.h

```diff
@@ -52,6 +53,23 @@ namespace tf2
     return std::chrono::system_clock::now();
   }
 
+  inline Duration durationFromSec(double t)
+  {
+    uint32_t sec, nsec;
+    sec = (uint32_t)floor(t);
+    nsec = (uint32_t)std::round((t-sec) * 1e9);
```

Comment:
I guess `round` actually makes sense here. It would still be nice to check the negative behavior in a test. My concern is that the result of `std::round((t-sec) * 1e9)` might be negative, which we are then just casting to an unsigned integer.